components/director/internal/domain/fetchrequest/service.go

```diff
@@ -75,38 +83,85 @@ func (s *service) fetchAPISpec(ctx context.Context, fr *model.FetchRequest) (*st
 	if resp.StatusCode != http.StatusOK {
 		errMsg := fmt.Sprintf("While fetching API Spec status code: %d", resp.StatusCode)
 		log.C(ctx).Errorf(errMsg)
-		return nil, s.fixStatus(model.FetchRequestStatusConditionFailed, str.Ptr(fmt.Sprintf("While fetching API Spec status code: %d", resp.StatusCode)))
+		return nil, FixStatus(model.FetchRequestStatusConditionFailed, str.Ptr(fmt.Sprintf("While fetching API Spec status code: %d", resp.StatusCode)), s.timestampGen())
 	}
 
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		log.C(ctx).WithError(err).Errorf("An error has occurred while reading API Spec.")
-		return nil, s.fixStatus(model.FetchRequestStatusConditionFailed, str.Ptr(fmt.Sprintf("While reading API Spec: %s", err.Error())))
+		log.C(ctx).WithError(err).Errorf("An error has occurred while reading Spec.")
+		return nil, FixStatus(model.FetchRequestStatusConditionFailed, str.Ptr(fmt.Sprintf("While reading API Spec: %s", err.Error())), s.timestampGen())
 	}
 
 	spec := string(body)
-	return &spec, s.fixStatus(model.FetchRequestStatusConditionSucceeded, nil)
+	return &spec, FixStatus(model.FetchRequestStatusConditionSucceeded, nil, s.timestampGen())
 }
 
 func (s *service) validateFetchRequest(fr *model.FetchRequest) error {
 	if fr.Mode != model.FetchModeSingle {
 		return apperrors.NewInvalidDataError("Unsupported fetch mode: %s", fr.Mode)
 	}
 
-	if fr.Auth != nil {
-		return apperrors.NewInvalidDataError("Auth for Fetch Request was provided, currently it's unsupported")
-	}
-
 	if fr.Filter != nil {
 		return apperrors.NewInvalidDataError("Filter for Fetch Request was provided, currently it's unsupported")
 	}
+
 	return nil
 }
 
-func (s *service) fixStatus(condition model.FetchRequestStatusCondition, message *string) *model.FetchRequestStatus {
+func (s *service) requestWithCredentials(ctx context.Context, fr *model.FetchRequest) (*http.Response, error) {
+	if fr.Auth.Credential.Basic == nil && fr.Auth.Credential.Oauth == nil {
+		return nil, apperrors.NewInvalidDataError("Credentials not provided")
+	}
+
+	req, err := http.NewRequest(http.MethodGet, fr.URL, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var resp *http.Response
+	if fr.Auth.Credential.Basic != nil {
+		req.SetBasicAuth(fr.Auth.Credential.Basic.Username, fr.Auth.Credential.Basic.Password)
+
+		resp, err = s.client.Do(req)
+
+		if err == nil && resp.StatusCode == http.StatusOK {
+			return resp, nil
+		}
+	}
+
+	if fr.Auth.Credential.Oauth != nil {
+		resp, err = s.secureClient(ctx, fr).Do(req)
+	}
+
+	return resp, err
+}
+
+func (s *service) secureClient(ctx context.Context, fr *model.FetchRequest) *http.Client {
```

Comment:
this http client creation is not really clean code, starting from a config, then you have to pass your existing client through the context so its transport is reused. But the clients timeout is not reused and thats why you set it explicitly. And not only that but the client method is on the Config struct... I normally try to avoid this package.
Maybe if you like to keep using the package you can use only the oauth2.TokenSource and the oauth2.Transport